### PR TITLE
[HOSTEDCP-1349 Prow CI workflow Enablement] Use MCE create HostedCluster on BM DualStack disconnected + agent

### PR DESCRIPTION
**Pull request by LiangquanLi930**
https://issues.redhat.com/browse/HOSTEDCP-1269 
Use MCE create HostedCluster on BM DualStack disconnected + agent 
testjob: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/46835/rehearse-46835-periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-dualstack-conformance/1738056578952597504 
```
[root@r640-01 ~]# oc get pod -A | grep -v "Running"
NAMESPACE                                          NAME                                                      READY   STATUS    RESTARTS      AGE
[root@r640-01 ~]# oc get node
NAME                   STATUS   ROLES    AGE   VERSION
ostest-extraworker-0   Ready    worker   45m   v1.27.8+4fab27b
ostest-extraworker-1   Ready    worker   45m   v1.27.8+4fab27b
ostest-extraworker-2   Ready    worker   45m   v1.27.8+4fab27b
[root@r640-01 ~]# oc get clusterversion version -ojsonpath="{.status.desired}" | jq
{
  "image": "virthost.ostest.test.metalkube.org:5000/localimages/local-release-image@sha256:eaec115c6b93ca0022c6d49898c1b2f95c1784fcca2d380c0423631c87bbce0e",
  "version": "4.14.0-0.ci-2023-12-21-213111"
}

```

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-dualstack-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-dualstack-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-ipv6-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-dualstack-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-dualstack-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-dualstack-conformance

File: ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.14-periodics.yaml [hunk 1]
```diff
@@ -93,11 +175,12 @@ periodics:
   labels:
     ci-operator.openshift.io/cloud: equinix-ocp-metal
     ci-operator.openshift.io/cloud-cluster-profile: equinix-ocp-metal
+    ci-operator.openshift.io/cluster: build05
     ci-operator.openshift.io/variant: periodics
     ci.openshift.io/generator: prowgen
     job-release: "4.14"
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
-  name: periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-conformance
+  name: periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-ipv6-conformance
```
Comment (jparrill) [inline ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.14-periodics.yaml:183]: Should all of these `ipv6` references to be something like dual? (if this is just the name it's fine, I just wondering that maybe we missed to move this from `ipv6` to `dual`)

File: ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.14-periodics.yaml [hunk 2]
```diff
@@ -106,8 +189,8 @@ periodics:
       - --lease-server-credentials-file=/etc/boskos/credentials
       - --report-credentials-file=/etc/report/credentials
       - --secret-dir=/secrets/ci-pull-credentials
-      - --secret-dir=/usr/local/e2e-agent-ovn-mce-disconnected-conformance-cluster-profile
-      - --target=e2e-agent-ovn-mce-disconnected-conformance
+      - --secret-dir=/usr/local/e2e-agent-ovn-mce-disconnected-ipv6-conformance-cluster-profile
```
Comment (jparrill) [inline ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.14-periodics.yaml:192]: Same as above

File: ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.14-periodics.yaml [hunk 3]
```diff
@@ -124,7 +207,7 @@ periodics:
       - mountPath: /secrets/ci-pull-credentials
         name: ci-pull-credentials
         readOnly: true
-      - mountPath: /usr/local/e2e-agent-ovn-mce-disconnected-conformance-cluster-profile
+      - mountPath: /usr/local/e2e-agent-ovn-mce-disconnected-ipv6-conformance-cluster-profile
```
Comment (jparrill) [inline ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.14-periodics.yaml:210]: Same as above

File: ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.14-periodics.yaml [hunk 4]
```diff
@@ -106,8 +189,8 @@ periodics:
       - --lease-server-credentials-file=/etc/boskos/credentials
       - --report-credentials-file=/etc/report/credentials
       - --secret-dir=/secrets/ci-pull-credentials
-      - --secret-dir=/usr/local/e2e-agent-ovn-mce-disconnected-conformance-cluster-profile
-      - --target=e2e-agent-ovn-mce-disconnected-conformance
+      - --secret-dir=/usr/local/e2e-agent-ovn-mce-disconnected-ipv6-conformance-cluster-profile
+      - --target=e2e-agent-ovn-mce-disconnected-ipv6-conformance
```
Comment (jparrill) [inline ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.14-periodics.yaml:193]: Same as above

File: ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.14-periodics.yaml (hunk 1 above)
Comment (LiangquanLi930) [reply ci-operator/jobs/openshift/hypershift/openshift-hypershift-release-4.14-periodics.yaml:183]: What does the `references` mean here? Here I just add `ipv6` which is using general workflow: hypershift-mce-agent-disconnected-conformance. For dualstack, i updated the `IP_STACK` inside.

**Pull request merged**